webui/react/src/hooks/useModal/UserSettings/useModalCreateGroup.tsx

```diff
@@ -121,31 +126,31 @@ const ModalForm: React.FC<Props> = ({ form, users, group, groupRoles }) => {
         </Form.Item>
       )}
       {rbacEnabled && canModifyPermissions && group && (
-        <>
-          <Form.Item label={GROUP_ROLE_LABEL} name={GROUP_ROLE_NAME}>
-            <Select
-              mode="multiple"
-              optionFilterProp="children"
-              placeholder={'Add Roles'}
-              showSearch>
-              {Loadable.match(roles, {
-                Loaded: (roles) =>
-                  roles.map((r) => (
+        <Spinner conditionalRender spinning={roles === undefined}>
+          <>
+            <Form.Item label={GROUP_ROLE_LABEL} name={GROUP_ROLE_NAME}>
+              <Select
+                mode="multiple"
+                optionFilterProp="children"
+                placeholder={'Add Roles'}
+                showSearch>
+                {roles // This is ok as the Spinner has conditionalRender active
+                  ? roles.map((r) => (
                     <Select.Option
                       disabled={groupRoles?.find((gr) => gr.id === r.id)?.fromWorkspace?.length}
                       key={r.id}
                       value={r.id}>
                       {r.name}
                     </Select.Option>
-                  )),
-                NotLoaded: () => undefined, // TODO show spinner when this is loading
```

Comment:
This should be blocked on Select having a loading state. We don't want to replace the field with a spinner.